server/valgrind-out.txt

```diff
@@ -0,0 +1,152 @@
+--29667-- REDIR: 0x48fc6b0 (libc.so.6:strncat) redirected to 0x48311d0 (_vgnU_ifunc_wrapper)
+--29667-- REDIR: 0x49e5500 (libc.so.6:__strrchr_avx2) redirected to 0x483ea10 (rindex)
+--29667-- REDIR: 0x48f70e0 (libc.so.6:malloc) redirected to 0x483b780 (malloc)
+--29667-- REDIR: 0x48f8b10 (libc.so.6:calloc) redirected to 0x483dce0 (calloc)
+--29667-- REDIR: 0x49e5310 (libc.so.6:__strchrnul_avx2) redirected to 0x4843540 (strchrnul)
+--29667-- REDIR: 0x49e56d0 (libc.so.6:__strlen_avx2) redirected to 0x483ef40 (strlen)
+--29667-- REDIR: 0x48f76d0 (libc.so.6:free) redirected to 0x483c9d0 (free)
+--29667-- REDIR: 0x49e86e0 (libc.so.6:__memcpy_avx_unaligned_erms) redirected to 0x48429f0 (memmove)
+--29667-- REDIR: 0x48fcfa0 (libc.so.6:__GI_strstr) redirected to 0x4843ca0 (__strstr_sse2)
+--29667-- REDIR: 0x49e1cc0 (libc.so.6:__memcmp_avx2_movbe) redirected to 0x48421e0 (bcmp)
+--29667-- REDIR: 0x49e0bd0 (libc.so.6:__strcmp_avx2) redirected to 0x483fed0 (strcmp)
+--29667-- REDIR: 0x48f7e80 (libc.so.6:realloc) redirected to 0x483df30 (realloc)
+==29667== Warning: ignored attempt to set SIGKILL handler in sigaction();
+==29667==          the SIGKILL signal is uncatchable
+--29667-- REDIR: 0x49e8b60 (libc.so.6:__memset_avx2_unaligned_erms) redirected to 0x48428e0 (memset)
+--29667-- REDIR: 0x49e86c0 (libc.so.6:__mempcpy_avx_unaligned_erms) redirected to 0x4843660 (mempcpy)
+--29667-- REDIR: 0x49e6200 (libc.so.6:__strncat_avx2) redirected to 0x483ede0 (strncat)
+==29667== Invalid free() / delete / delete[] / realloc()
+==29667==    at 0x483CA3F: free (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
+==29667==    by 0x1095B8: signal_handler (aesdsocket.c:60)
+==29667==    by 0x48A008F: ??? (in /usr/lib/x86_64-linux-gnu/libc-2.31.so)
+==29667==    by 0x497D0D4: accept (accept.c:26)
+==29667==    by 0x109AA0: socket_connect (aesdsocket.c:207)
+==29667==    by 0x1096E6: main (aesdsocket.c:101)
+==29667==  Address 0x4c664a0 is 0 bytes inside a block of size 20,040 free'd
+==29667==    at 0x483CA3F: free (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
+==29667==    by 0x10A022: socket_connect (aesdsocket.c:332)
+==29667==    by 0x1096E6: main (aesdsocket.c:101)
+==29667==  Block was alloc'd at
+==29667==    at 0x483B7F3: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
+==29667==    by 0x109EDE: socket_connect (aesdsocket.c:308)
+==29667==    by 0x1096E6: main (aesdsocket.c:101)
+==29667== 
+==29667== 
+==29667== HEAP SUMMARY:
+==29667==     in use at exit: 0 bytes in 0 blocks
+==29667==   total heap usage: 277 allocs, 278 frees, 2,198,379 bytes allocated
+==29667== 
+==29667== All heap blocks were freed -- no leaks are possible
+==29667== 
+==29667== ERROR SUMMARY: 1 errors from 1 contexts (suppressed: 0 from 0)
+==29667== 
+==29667== 1 errors in context 1 of 1:
+==29667== Invalid free() / delete / delete[] / realloc()
```

Comment:
@ayswariya088 these are pointing out real problems which are related to freeing content in a signal handler.